### ISSUE TITLE
Request to Increase Extension Upload File Size Limit

### ISSUE THREAD

**Issue by Nick-1979** [mozilla/addons-linter#4942]
Dear Firefox Development Team,

I am writing to inform you of an issue I encountered while attempting to update my extension on Firefox. Unfortunately, despite successfully updating the extension on Chrome and Edge with new features, I am currently unable to do the same on Firefox due to the existing file size limitation.

The error message states that the file size is too large to parse.

I kindly request that you consider increasing the file size limit for extension updates on Firefox. This adjustment would allow developers like me to provide the same enhanced functionality to Firefox users that is already available on other browsers.

Thank you for your attention to this matter. I greatly appreciate the efforts of the Firefox development team and hope for a positive resolution.

<img width="1040" alt="Screenshot 2023-07-02 at 01 17 54" src="https://github.com/mozilla/addons-linter/assets/46442452/3f0c133b-c781-48e3-8224-1defea7ab64c">
<img width="1138" alt="Screenshot 2023-07-02 at 01 18 14" src="https://github.com/mozilla/addons-linter/assets/46442452/a0eea7ed-b2ef-436b-ad9d-0b3842a234b5">

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/ADDLINT-211)

**Comment by diox** [comment]
Hi,

This has already been reported in https://github.com/mozilla/addons-linter/issues/4748 for a specific case - [`tesseract.js`](https://github.com/naptha/tesseract.js).

We have to have some kind of limit to avoid abuse as parsing and analyzing submitted code is time consuming and costly as the size increases.

What limit would work in your case ?

**Comment by Nick-1979** [comment]
Hi @diox,

Thank you for responding. In my situation, a file size of 5 MB is acceptable.

**Comment by diox** [comment]
I'm not opposed to a bump to 5 MB - there are several developers that it would benefit and it's been a while since we last increased that limit - but we need to make some internal architecture changes in our private scanners before considering this.

**Comment by Nick-1979** [comment]
Hello,

I would like to know if there is any progress on resolving this issue or any estimate on when we can update our extensions without encountering this problem. 

Thank you for your hard work and attention.

**Comment by diox** [comment]
Hi,

Sorry for not answering sooner. This is still blocked on migrating infrastructure of our private scanners but we intend to unblock that in January (wanted to do this earlier but we ran into issues when migrating the bulk of AMO infrastructure that pushed this further and now it's too late, we're entering a code/infra freeze for end-of-year holidays). Once this is done and stable there should be no hard blockers left and we should be able to bump that limit.

**Comment by Nick-1979** [comment]
Hi @diox

I would appreciate it if you could provide an update on this matter.

Thank you!

**Comment by Nick-1979** [comment]
I'm disappointed to see that this issue is still ongoing after a year. While I understand these problems can be complex to resolve, I had hoped more progress would have been made by now. 
I encourage the team to continue working diligently to address this matter in a timely manner. I remain hopeful that a solution can be found soon. 

Thank you for your continued efforts on this important matter.

<img width="1044" alt="Screenshot 2024-06-25 at 17 26 20" src="https://github.com/mozilla/addons-linter/assets/46442452/7d733b13-9d08-4c77-80ef-3a1124e2f97a">

**Comment by mhabsaoui** [comment]
Hi, 

Same here :-/
We just added new features to our addon [Click&Read](https://addons.mozilla.org/fr/firefox/addon/click-and-read/), and the `Options.js` jumps from 3MB to 4.99MB.

And the limit is 4MB !

Please help us because Mozilla Store on late review deactivated all our addon versions, and asked us to add a Privacy Consent ASAP. We did and now this...

**Comment by mhabsaoui** [comment]
OK, if I am right?, according to Firefox Doc. we can submit our raw code base, as is in our git repository (not builded nor minified, but with building instructions and info).

This should do because the review will pass the file size check, as our files are in distinct modules at start...

https://extensionworkshop.com/documentation/publish/source-code-submission/

**Comment by Nick-1979** [comment]
@mhabsaoui 

"When you **upload your extension** to AMO, you will need to **provide your source code** and instructions for building that source code." This means you need to upload your code **after** uploading the build if it is minified or using webpack.

**Comment by mhabsaoui** [comment]
> @mhabsaoui
> 
> "When you **upload your extension** to AMO, you will need to **provide your source code** and instructions for building that source code." This means you need to upload your code **after** uploading the build if it is minified or using webpack.

Yes, indeed just checked with my collegue, first we upload the dist zip, and in case of build tool (our case with Webpack), then we also upload the SRC code zip !

Then, what can we do, only to try to split our features in others Views...

Just don't see why :

- we are limited in code size for new features ?
- why the addon linter isn't applied on raw source code, insted of build dist ?

Thanks

**Comment by mhabsaoui** [comment]
After some digging and assets folder refactoring, the Prod mode does its job and the output `options.js` file is back to normal size (1,3 MB).
Don't know for sure if it's realted to the fact of upgradin Webpack and the `args.mode` usage in its config file...

**Comment by dotproto** [comment]
> why the addon linter isn't applied on raw source code, insted of build dist ?

I see two reasons.

1. The linter is run against the extension code submitted by the developer because that is the code that end users will be installing and running in their browsers.
2. The original languages used to write the extension may not be JavaScript, JSON, etc. "Compile-to-JS" languages are common and it's infeasible for the linter to support all of them.

**Comment by ThienNguyen1410** [comment]
i see

**Comment by qqzm** [comment]
It would make my build process so much easier if you could increase this limit to 15MB.

**Comment by darioAnongba** [comment]
@diox, any update on that increase to 5MB (or more)? I see many posts complaining about this issue for many years now. 

Chrome does not have this problem and their Extension Store is working fine.Do you have this limitation because of a limit on resources?

**Comment by qqzm** [comment]
My issue is with a single JSON data file in my extension. The file is a little over 10MB.

My solution is to package 4MB of data with the extension and then download the full 10MB file in the background on first launch. But this adds complexity to my build process as I have to compile a full copy of the extension for Chrome, Opera and Edge, then chop the file down to 4MB and create a separate package just for Firefox.

**Comment by Juraj-Masiar** [comment]
@qqzm another workaround could be to zip it with `gzip` and then use the new Compression Screams API to unzip it at runtime:
https://developer.mozilla.org/en-US/docs/Web/API/Compression_Streams_API

**Comment by qqzm** [comment]
> [@qqzm](https://github.com/qqzm) another workaround could be to zip it with `gzip` and then use the new Compression Screams API to unzip it at runtime: https://developer.mozilla.org/en-US/docs/Web/API/Compression_Streams_API

Thanks for your suggestion. Unfortunately, zipping it does not get it below 4MB.

**Comment by qqzm** [comment]
> > [@diox](https://github.com/diox), any update on that increase to 5MB (or more)? I see many posts complaining about this issue for many years now.
> 
> We have made the migrations we wanted to unblock this, so I'll look into updating the limit to 5MB shortly.

Please can you consider making it 15MB for JSON files.
An increase to 5MB would make no difference to me. The file I have problems with is already over 10MB.

**Comment by SunsetMkt** [comment]
This may also affect Bitwarden Firefox Addon [v2025.5.1](https://github.com/bitwarden/clients/releases/tag/browser-v2025.5.1). In my custom build, a `js` file is 4~5MB and cannot pass the lint.

**Comment by ssfeatures** [comment]
Thank you @diox!

**Comment by ioanarusiczki** [comment]
@diox I tried on linter 7.14.0 (I didn't have a valid js file) but I could check the issue increasing a file's size 

Above 4MB a js. file would throw the error on linter 7.13.0 ->  FILE_TOO_LARGE
On 7.14.0 I got JS_SYNTAX_ERROR (expected since it's not valid) for js. files between 4-5 MB. And above 5MB would throw FILE_TOO_LARGE

And I didn't have any warning/errors with a larger css file ~ 10 Mb
